http-client/Network/HTTP/Client/Types.hs

```diff
@@ -820,7 +824,20 @@ data ManagerSettings = ManagerSettings
     -- Default: respect the @proxy@ value on the @Request@ itself.
     --
     -- Since 0.4.7
-    , managerMaxHeaderLength :: Maybe MaxHeaderLength
+    , managerMaxHeaderLength :: MaxHeaderLength
+    -- ^ Configure the maximum size, in bytes, of an HTTP header field.
+    -- Set it to 0 to remove this limit (eg: for debugging purposes).
```

Comment:
I don't think it's necessary to use 0 as a sentinel value. `maxBound :: Word32` is more than 4 GB, and `maxBound :: Word64` is unfathomably large. In other words, using `maxBound` would effectively disable these checks anyway.

Reply:
I agree, and this is why I just removed the `n /= 0` condition from the previous predicates.